### PR TITLE
Normalize name field

### PR DESCRIPTION
**Pull request by trodge**
If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
monitoring: fixed an issue which occurred when `name` field of `google_monitoring_monitored_project` was long-form
```

### CODE REVIEW

**Comment by zli82016** [comment]
Which test has the problem, @trodge ?  Is this one TestAccMonitoringMonitoredProject_monitoringMonitoredProjectBasicExample?

**Comment by zli82016** [comment]
Add a secondary reviewer @slevenick

File: mmv1/products/monitoring/MonitoredProject.yaml [hunk 1]
```diff
@@ -31,6 +31,7 @@ description: "A [project being monitored](https://cloud.google.com/monitoring/se
 immutable: true
 error_retry_predicates: ['transport_tpg.IsMonitoringPermissionError']
 custom_code: !ruby/object:Provider::Terraform::CustomCode
+  custom_import: templates/terraform/custom_import/monitoring_monitored_project.go.erb
```
Comment (slevenick) [inline mmv1/products/monitoring/MonitoredProject.yaml:34]: Why do we need a custom import here? Is it only to support importing with the `v1/` prefix?

File: mmv1/products/monitoring/MonitoredProject.yaml [hunk 2]
```diff
@@ -17,7 +17,7 @@ base_url: v1/locations/global/metricsScopes
 create_url: v1/locations/global/metricsScopes/{{metrics_scope}}/projects
 delete_url: v1/locations/global/metricsScopes/{{metrics_scope}}/projects/{{name}}
 self_link: v1/locations/global/metricsScopes/{{metrics_scope}}
-id_format: v1/locations/global/metricsScopes/{{metrics_scope}}/projects/{{name}}
+id_format: locations/global/metricsScopes/{{metrics_scope}}/projects/{{name}}
```
Comment (slevenick) [inline mmv1/products/monitoring/MonitoredProject.yaml:20]: I think we need an upgrade state function to make sure any of these resources that were created with the `v1/` prefix are moved to the correct id format

**Review by slevenick — CHANGES_REQUESTED** [review]

File: mmv1/products/monitoring/MonitoredProject.yaml (hunk 1 above)
Comment (trodge) [reply mmv1/products/monitoring/MonitoredProject.yaml:34]: It's there to shorten the `name` and `metrics_scope` fields in the import id.

**Review by slevenick — APPROVED** [review]
LGTM, I tried out the upgrade path from 4.70.0 --> 4.73.1 --> this version manually and it works

**Pull request merged**